common/auth_util.go

```diff
@@ -26,12 +26,35 @@ import (
 )
 
 const (
-	resourceURNTemplateCreateDestination = "urn:cherami:dst:%v"
+	resourceURNTemplateCreateDestination = "urn:cherami:dst:%v:%v"
 )
 
-// GetResourceRootURN returns the root resource URN, e.g. urn:cherami:dst:zone1_prod
+// GetResourceURNCreateDestination returns the resource URN to create destination, e.g. urn:cherami:dst:zone1_prod:/prefix1
 // We use URN (Uniform Resource Name) like this: https://www.ietf.org/rfc/rfc2141.txt
-func GetResourceRootURN(scommon SCommon) string {
+func GetResourceURNCreateDestination(scommon SCommon, dstPath *string) string {
+	var dstPathString string
+	if dstPath == nil {
+		dstPathString = ""
+	} else {
+		dstPathString = getPathRootName(dstPath)
+	}
 	deploymentName := scommon.GetConfig().GetDeploymentName()
-	return fmt.Sprintf(resourceURNTemplateCreateDestination, strings.ToLower(deploymentName))
+	return fmt.Sprintf(resourceURNTemplateCreateDestination, strings.ToLower(deploymentName), strings.ToLower(dstPathString))
+}
+
+func getPathRootName(path *string) string {
+	if path == nil {
+		return ""
+	}
+
+	parts := strings.Split(*path, "/")
+	if len(parts) == 0 {
```

Comment:
I don't think this condition will ever be satisfied. Even for empty string, len(parts) will be 1